go/service/main.go

```diff
@@ -102,16 +108,21 @@ func (d *Service) Handle(c net.Conn) {
 		logReg = newLogRegister(d.logForwarder, d.G().Log)
 		defer logReg.UnregisterLogger()
 	}
-
-	if err := d.RegisterProtocols(server, xp, connID, logReg, d.G()); err != nil {
+	shutdowners, err := d.RegisterProtocols(server, xp, connID, logReg, d.G())
+	defer func() {
```

Comment:
When the service shuts down, does it make it into this function before the service stops?

With the hook that was there before, it was called by GlobalContext.Shutdown.  Obviously it was broken in that it didn't stop when clients disconnected, but just want to make sure that in the case of service shutdown (and not client disconnect) the shutdown function is being called.